### ISSUE TITLE
Shortest path is not the shortest

### ISSUE THREAD

**Issue by dNaszta** [dominikbraun/graph#88]
```
func TestShortestPath(t *testing.T) {
	g := graph.New(graph.StringHash, graph.Weighted(), graph.Directed())

	g.AddVertex("start")
	g.AddVertex("a")
	g.AddVertex("b")
	g.AddVertex("fin")

	g.AddEdge("start", "a", graph.EdgeWeight(6))
	g.AddEdge("start", "b", graph.EdgeWeight(2))
	g.AddEdge("a", "fin", graph.EdgeWeight(1))
	g.AddEdge("b", "a", graph.EdgeWeight(3))
	g.AddEdge("b", "fin", graph.EdgeWeight(5))

	path, _ := graph.ShortestPath(g, "start", "fin")

	wanted := []string{"start", "b", "a", "fin"}
	if !reflect.DeepEqual(path, wanted) {
		t.Errorf("shortest path failed, result: %v, wanted: %v", path, wanted)
	}
}
```

shortest path failed, result: [start b fin], wanted: [start b a fin]

**Comment by dominikbraun** [comment]
Nice catch, thank you.

![issue-88](https://user-images.githubusercontent.com/36575275/227789319-0b4074f0-aaaa-43a5-b104-a8abdc2bceaf.jpg)

**Comment by dominikbraun** [comment]
The fix has been released in [graph v0.16.2](https://github.com/dominikbraun/graph/releases/tag/v0.16.2).

**Comment by dominikbraun** [comment]
The fix has been released in [v0.16.2](https://github.com/dominikbraun/graph/releases/tag/v0.16.2).
